web/client-api/src/main/java/io/deephaven/web/client/api/WorkerConnection.java

```diff
@@ -698,22 +700,12 @@ public void onError(Throwable throwable) {
 
     @SuppressWarnings({"unchecked", "rawtypes"})
     public Promise<Object> getObject(JsVariableDefinition definition) {
-        switch (VariableType.valueOf(definition.getType())) {
-            case Table:
-                return (Promise) getTable(definition, null);
-            case TreeTable:
-                return (Promise) getTreeTable(definition);
-            case Figure:
-                return (Promise) getFigure(definition);
-            case Pandas:
-                return (Promise) getPandas(definition);
-            // case OtherWidget:
-            // return (Promise) getWidget(definition.getName());
-            // case TableMap:
-            // return (Promise) getTableMap(definition.getName());
-            default:
-                return Promise.reject(
-                        new Error("Object " + definition.getTitle() + " unknown type " + definition.getType() + "."));
+        if (definition.getType().equals(VariableType.Table.toString())) {
+            return (Promise) getTable(definition, null);
+        } else if (definition.getType().equals(VariableType.Figure.toString())) {
```

Comment:
Can we use JsVariableChanges.FIGURE instead, so we know the string is the same that JS might have passed in?

Reply:
Yup makes sense